core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -231,26 +237,17 @@ public X509Certificate getTrustedCertificateAuthority() throws CertificateExcept
      * Once a (non {@code null}) trust anchor has been created, it will be cached and
      * returned on subsequent invocations of this method.
      * 
+     * @deprecated Replaced by {@link #getTrustAnchors()}
      * @return The trust anchor or {@code null} if no trusted certificate authority
      *         has been set.
      * @throws GeneralSecurityException if the value of the <em>trusted-ca</em> property
-     *             cannot be parsed into a trust anchor, e.g. because of unsupported
-     *             key type, malformed certificate or public key encoding etc. 
+     *              cannot be parsed into a trust anchor, e.g. because of unsupported
+     *              key type, malformed certificate or public key encoding etc.
      */
+    @Deprecated
     @JsonIgnore
     public TrustAnchor getTrustAnchor() throws GeneralSecurityException {
-
-        if (trustAnchor != null) {
-            return trustAnchor;
-        } else {
-            final X509Certificate cert = getTrustedCertificateAuthority();
-            if (cert != null) {
-                trustAnchor = new TrustAnchor(cert, null);
-                return trustAnchor;
-            } else {
-                return getTrustAnchorForPublicKey(getProperty(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA));
-            }
-        }
+        return (getTrustAnchors() == null ? null : getTrustAnchors().get(0));
```

Comment:
IMHO we should remove this method.

Reply:
as i mention in a previous comment, we can keep this for the current content model

Reply:
as I mentioned above, should probably return `List<TrustAnchor>`.